bin/ember-template-lint.js

```diff
@@ -419,8 +419,9 @@ async function run() {
     let hasWarnings = results.warningCount > 0;
     let hasTodos = options.includeTodo && results.todoCount;
     let hasUpdatedTodos = options.updateTodo;
+    let alwaysEmitOutputFile = options.format === 'sarif' && options.outputFile;
```

Comment:
Why do we do the SARIF check here? Is that because we haven't migrated the JSON formatter to use the new infrastructure yet?

Reply:
I kinda think it should always be up to the formatter to decide if we should emit. The default one (pretty) could choose to print nothing if there were no results (the current behavior), but then it would be trivially possible to author a formatter that behaved like our built in "file emitting" ones. 

What do you think?

Reply:
Ya I like this. So we're essentially saying removing the conditional check and letting formatters make the call. We can also populate these checks above for the options passed to formatters, for convenience.

Reply:
OK I've moved this check into the formatter itself, in addition to the checks in all the respective formatters.